### ISSUE TITLE
qml: update checker/notifications (in-app)

### ISSUE THREAD

**Issue by SomberNight** [spesmilo/electrum#8311]
I think it would be useful for the qml gui to have a built-in update checker, similar to the one in the desktop qt gui (see [`gui/qt/update_checker.py`](https://github.com/spesmilo/electrum/blob/75578d2baab4675a0123784797eaac381151f303/electrum/gui/qt/update_checker.py)). 

In particular, I would like us to have a mechanism for notifying Android users of updates independent of the google play store. Consider e.g. there is a security fix but there is some issue with publishing on google play. For example, review times might be long, or there might be some new restrictions imposed on updates (e.g. a new target sdk version we cannot update to in time). Atm we can release a new apk on the website but we have no way of notifying users of there being an update.

-----

One thing to note is that we might want to make the current version file more generic, relatedly.
E.g. [current version file](https://github.com/spesmilo/electrum-web/blob/58755abe49fcbcf295aed67ae3c3d52c1902b07b/version):
```
{
    "version":"4.4.0",
    "signatures": {
        "13xjmVAB1EATPP8RshTE8S8sNwwSUM9p1P": "H0ux4qW9K7ROuwZlvvp+/hUnqBYXCQS85yQPJadaT+7iaxS8aplTxm/+KpTUWjWInrwxj9Ni3kbk3/F+aWaRi6Q="
    }
}
```
In a typical release flow, we upload binaries to the website and update the `version` file at the same time. We also publish the android apk on the google play store roughly at the same time, however it takes several days for "review" on google play to finish before the apk becomes available there. During this "review" phase, typically we would probably not want users to see a "new update" notification (we would want to capability of showing it but would not want to show it in most cases). Also, sometimes we do "staged rollouts" in which case again we probably don't want users to see notifications.

We could extend the version file json to add more capabilities. I can flesh out a more concrete proposal for that part if there is interest.

**Comment by accumulator** [comment]
Good idea. As a very simple addition we could add the release date. We can then tune the moment we show a notification in the android app, to be sure a phased rollout is complete.
